### PR TITLE
Feature/#27 leaderboard

### LINKED ISSUE #27: [MAIN] [Pipeline] Model tracking leaderboard
Writes to singular csv which tracks the performance/evaluation metrics of the models and location at which each model is saved.

Potentially also track:
- hyperparameters used to train it -> useful for side-by-side comparisons in fine-tuning stage 
- pipeline specifics (non-hyperparameter selections made such as model architecture, loss function selection, etc.) 
- the pipeline version (referencing exact point in repository timeline) -> useful if large changes to overall pipeline are made

### PR DESCRIPTION
**Pull request by AndreHrs**
# WARNING
Dependent on PR #85
This is a chain of PR that should be merged sequentially PR #84  -> PR #85 -> This PR
---

Finished the leaderboard feature. Tracks metrics mentioned in the feature #27 
Passed the leaderboard to training loop runner with configurable parameter
```
log_leaderboard=True, <-- Flag to log the run or not, defaults to True 
leaderboard_dir="leaderboard", <-- The directory for where the "leaderboard.csv" will be stored
```

Identity of the runner will be tracked via those resolution priorities:
1. git config user.name  (or user.email as fallback)
2. if 1 is not present, use their device name

After identity is resolved, owner.conf (gitignored) will be cached in the leaderboard directory.

### CODE REVIEW

**Review by Mars-Martiny — APPROVED** [review]
Great work adding in the leaderboard!
Its great that the idea of using the github username to track the owners of the runs worked well. 
This will have to be adjusted slightly, perhaps with a secondary version of the leaderboard csv, to account for the changes planned for the evaluation metrics (class specific metrics).
